### ISSUE TITLE
pythonw.exe is not working on win10, and thus scrobbler is not running in background.

### ISSUE THREAD

**Issue by Umair444** [iamkroot/trakt-scrobbler#63]
**Describe the bug**
Error happens sometimes while sending data to API, but the main problem is that it's not running in background. After executing "trakts start", program 'python.exe' starts and it remains running (from Task Manager) until I close the cmd. I think this happens because pythonw,exe is not running...

**Desktop (please complete the following information):**
 - OS: [e.g. Windows 10]
 - Python Version: [e.g. 3.8.2]
 - Player [e.g. VLC]
 - Player Version [e.g. 3.0.10]

**To Reproduce**
1. Open '....' file
2. Play till '....'
3. Close player/pause it
4. See error in log file

**Log file**
<details><summary>Click to see log contents</summary> 
<p>

```
2020-05-08 21:57:03,400 - DEBUG - scrobbler - scrobbler - Progress: 0.26%
2020-05-08 21:57:03,400 - DEBUG - scrobbler - trakt_interface - Searching cache.
2020-05-08 21:57:03,400 - DEBUG - scrobbler - trakt_interface - {'movie': {'ids': {'trakt': 383525}}}
2020-05-08 21:57:04,090 - INFO - scrobbler - scrobbler - Scrobble stop successful for Portrait of a Lady on Fire
2020-05-08 21:57:04,636 - INFO - scrobbler - scrobbler - Scrobble stop successful for Portrait of a Lady on Fire

```
</p>
</details>

**Comment by Umair444** [comment]
Well, I think after a lot of trouble and reading some basic info of python; I did it by replacing the **'python.exe'** in `.local\pipx\venvs\trakt-scrobbler\Scripts\trakts.exe` to **'pythonw.exe'**. After a restart auto-start works fine, and scrobbler is also working perfect. Still, I'm not closing the issue, as maybe some other better solution could exist...

**Comment by iamkroot** [comment]
I don't exactly understand your error here.  Is it that after you run `trakts start`, your command prompt cannot be closed? Or that it opens up a new window with python in it?

**Comment by Umair444** [comment]
> I don't exactly understand your error here. Is it that after you run `trakts start`, your command prompt cannot be closed? Or that it opens up a new window with python in it?

With `trakts start`, python.exe starts, and we can check it under task manager details tab, but as soon as I turn off the command prompt, it kills the `python.exe`, and thus `trakts.exe`. 
So, to run the scrobbler I have to keep my cmd open, and execute the `start` command after every restart.

**Comment by iamkroot** [comment]
That is definitely weird. How exactly did you install python in your system?

**Comment by iamkroot** [comment]
Also, do you mean to say that `pythonw.exe` did not exist at first, and by renaming `python.exe` to `pythonw.exe`, everything was fixed (i.e. you could close cmd without killing the scrobbler)?

**Comment by Umair444** [comment]
> That is definitely weird. How exactly did you install python in your system?

Install it from .exe from official site..

**Comment by Umair444** [comment]
I don't know why but with the pipx method, my installation project files are in: `C:\Users\mugha\.local\pipx\venvs\trakt-scrobbler` while the other users have them in `AppData\Local`;  where my python is installed. 
So, I tried to delete the folder manually with the yaml file, to reinstall with the given procedure; it didn't fix the error; but now my scrobbler is also not working... 

Here's the log:
```
{'Content-Type': 'application/json', 'trakt-api-key': 'ab0133a2365b2c64d70fd2adf3c7e775a4131471b56340933335af1b94785a3a', 'trakt-api-version': '2', 'Authorization': 'Bearer a6a401ff592a6c565a0051dc75b382e25e14ea08b958678b1fe4c3a793b3fc95'}, 'json': {'show': {'ids': {'trakt': 160674}}, 'episode': {'season': 1, 'number': 11}, 'progress': 9.27}}
2020-05-10 00:25:54,878 - **DEBUG - scrobbler - utils - Response: <Response [404]>** 
2020-05-10 00:25:54,878 - **WARNING - scrobbler - scrobbler - Scrobble unsuccessful.**
2020-05-10 00:26:04,519 - DEBUG - scrobbler - scrobbler - Progress: 0.15%
2020-05-10 00:26:04,519 - DEBUG - scrobbler - trakt_interface - Searching cache.
2020-05-10 00:26:04,519 - DEBUG - scrobbler - trakt_interface - {'show': {'ids': {'trakt': 160674}}, 'episode': {'season': 1, 'number': 11}}
2020-05-10 00:26:04,897 - **WARNING - scrobbler - utils - Request failed**
2020-05-10 00:26:04,897 - DEBUG - scrobbler - utils - Request: post {'url': 'https://api.trakt.tv/scrobble/start', 'headers': {'Content-Type': 'application/json', 'trakt-api-key': 'ab0133a2365b2c64d70fd2adf3c7e775a4131471b56340933335af1b94785a3a', 'trakt-api-version': '2', 'Authorization': 'Bearer a6a401ff592a6c565a0051dc75b382e25e14ea08b958678b1fe4c3a793b3fc95'}, 'json': {'show': {'ids': {'trakt': 160674}}, 'episode': {'season': 1, 'number': 11}, 'progress': 0.15}}
2020-05-10 00:26:04,898 - **DEBUG - scrobbler - utils - Response: <Response [404]>** 
2020-05-10 00:26:04,898 - **WARNING - scrobbler - scrobbler - Scrobble unsuccessful.**

```

**Comment by iamkroot** [comment]
> other users have them in <username>\AppData\Local\<...>

Not sure about that, even i have the pipx env inside .local folder in windows. That shouldn't affect the installation.

Maybe try wiping pipx, the trakt scrobbler environment and even the python installation. See if reinstalling from scratch fixes this

**Comment by iamkroot** [comment]
Are you sure you ran `pipx` and not `pip`?

**Comment by Umair444** [comment]
After uninstalling everything and deleting respective environmental variables: now it's fixed. This time (3rd) I installed the Python in `C:\Python` and keep everything else default. Well the problem was weird, maybe I did some mistake while installing, or maybe the problem was with the variables. 
Thanks for help.

**Comment by Umair444** [comment]
Also, last time I had 2 versions of Python installed; other was 2.7..

**Comment by iamkroot** [comment]
Hey @Umair444, I checked on a clean windows installation and saw that installing by `pipx` caused no `pythonw.exe` to be formed. It probably does some internal stuff to make sure no window is created. 

Closing this for now. If someone else also reports this, then I'll take a proper look.

**Comment by cyatek** [comment]
I got the same thing happening, once I close the cmd, every process stops. I'm guessing it's my fault since my python install is probably a mess, I tried reinstalling but without any success so I'm probably doing that wrong too. If it isn't asking too much @Umair444 could ELI5 how to do a clean install of everything?

**Comment by Umair444** [comment]
First try last 4 steps before reinstallation procedure. It worked for me, and could work for you too. 

Uninstall Completely:
1. Uninstall python from control panel
2. Delete all remaining file from installation directory 
3. Delete all environmental variables of python from path.
`win+R --> SystemPropertiesAdvanced --> Environment Variables --> System Variable --> Path`
4. Delete user data --> run %appdata% --> Delete **trakt-scrobbler** (contains yaml configuration files) and **Python** folder. 

Install Python:
1. Download [python 3](https://www.python.org/downloads/)from the official site
2. Install it in C. 
3. After installation get the installation directory path, which should be `C:\Python\Python38\`, and add this in environment variable path. 
4. Also, add `C:\Python\Python38\Scripts\` to prevent errors. 
5. Open cmd and execute `python` or `py -3` to check if windows is recognizing the path or not. 

Install trakt-scrobbler:
1. Install pipx `py -3 -m pip install --user pipx`
2. Use ensurepath command, to check if pipx can be used in cmd `py -3 -m pipx ensurepath`
3. If cmd is doesn't recognize `pipx`, try adding path manually to Envr Variable > Path 
`[C:\Users\<UserName>\AppData\Roaming\Python\Python38\site-packages\pipx]`
3. With the 'pipx', install trakt-scroobler -- `pipx install trakt-scrobbler`
4. Execute 'init', config the scrobbler and restart computer. 

If Still Not Work Try (NOT-RECOMMENDED):
1. C:\Users\<UserName>\.local\bin
2. Open `trakts.exe` in notepad
3. Search for python.exe, and change it to pythonw.exe, there should be a single python.exe in here.
4. Start scrobbler --> Close CMD --> Open Task Manager (CTRL+SHIFT+ESC) --> Details --> Search for trakts.exe, it should be running. Also, search python.exe. If with these steps worked for you, then there'll be pythonw.exe process instead of python.exe. 
**NOTE:** If you follow these steps you'll face a problem with `autostart`. So, after shutdown, the process will not start until you execute `trakts start` command.

**Comment by Umair444** [comment]
And you really asked for too much... 😁

**Comment by cyatek** [comment]
Yes! Thank you very much for your time :) I tried the last solution and it's working fine, I'll later attempt the clean install once the next update drops.
